src/pages/test-report/cat-be/test-report.cat-be.page.ts

```diff
@@ -187,23 +187,25 @@ export class TestReportCatBEPage extends BasePageComponent {
       isSeriousMode$.pipe(map(result => (this.isSeriousMode = result))),
       isDangerousMode$.pipe(map(result => (this.isDangerousMode = result))),
       manoeuvres$.pipe(map(result => (this.manoeuvresCompleted = result))),
-      // isLegalRequirementsValid$.pipe(
-      //   map(result => (this.isLegalRequirementsValid = result)),
-      // ),
       isEtaValid$.pipe(map(result => (this.isEtaValid = result))),
-      catBELegalRequirements$.pipe(
-        map(result => (this.catBELegalRequirements = result)),
+      testData$.pipe(
+        map((data) => {
+          this.isTestReportValid =
+            this.testReportValidatorProvider.isTestReportValid(data, TestCategory.BE);
+          this.missingLegalRequirements =
+            this.testReportValidatorProvider.getMissingLegalRequirements(data, TestCategory.BE);
+        }),
       ),
     ).subscribe();
   }
 
   onEndTestClick = (): void => {
     const options = { cssClass: 'mes-modal-alert text-zoom-regular' };
-    if (!this.isLegalRequirementsValid) {
+    if (!this.isTestReportValid) {
       this.modal = this.modalController.create(
         'LegalRequirementsModal',
```

Comment:
can we refactor this to use a page constant?

Reply:
done